packages/build/config/tsconfig.common.json

```diff
@@ -14,6 +14,7 @@
 
     "lib": ["es2018", "esnext.asynciterable"],
     "module": "commonjs",
+    "esModuleInterop": true,
```

Comment:
Quoting from https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-from-commonjs-modules-with---esmoduleinterop:

> Note: The new behavior is added under a flag to avoid unwarranted breaks to existing code bases. We highly recommend applying it both to new and existing projects. For existing projects, namespace imports (`import * as express from "express"; express();`) will need to be converted to default imports (`import express from "express"; express();`).

By enabling `esModuleInterop` in the shared tsconfig file, we are forcing this breaking change on all existing LoopBack projects. At minimum, I think this change should be released in a semver-major version of `@loopback/build`.

I have also a question about the impact of this change on code consuming LoopBack modules, especially when the consumer (the app project) and the producer (us, loopback-next) have different setting of `esModuleInterop`.

- Use case 1:

  The application is using an older version of LB4 packages, where the javascript files in `dist` were created using our current TypeScript config (`esModuleInterop` disabled); and uses the current shared TypeScript config.

  Now the application decides to enable `esModuleInterop`, but does not update LB packages. 

  What will happen? Will existing imports in the application continue to work?

- Use case 2:

  Same initial setup as above, but now the application decides to keep `esModuleInterop` disabled, but update LB packages to the latest versions, where `dist` is created using `esModuleInterop` enabled.

  What will happen? Will existing imports in the application continue to work?

Depending on the answer to these questions, we may need to bump semver-major version in all packages, in addition to `@loopback/build`.

If this change is backwards-compatible as far as the consumers of our packages are concerned, then we need to split the change in two commits: a semver-major change in `@loopback/build` as the first commit, and then a standalone "chore" or "refactor" (?) commit updating our codebase to accommodate the new compiler settings.